### PR TITLE
new format for ci config file

### PR DESCRIPTION
**Pull request by mockersf**
# Objective

- Current config file is hard to extend

## Solution

- Instead of an hard coded list of field, the file now has a list of `(frame, event)`, and will deal with know events (exiting or taking a screenshot), or send an event for others that can be dealt by third party plugins

### CODE REVIEW

File: crates/bevy_dev_tools/src/ci_testing.rs [hunk 1]
```diff
@@ -20,30 +19,26 @@ use serde::Deserialize;
...
-            app_exit_events.send(AppExit::Success);
-            info!("Exiting after {} frames. Test successful!", exit_after);
-        }
-    }
-    *current_frame += 1;
+/// Setup for a test.
+#[derive(Deserialize, Default)]
+struct CiTestingSetup {
+    /// The time in seconds to update for each frame.
+    pub frame_time: Option<f32>,
```
Comment (BD103) [inline crates/bevy_dev_tools/src/ci_testing.rs:35]: Could you be a bit more specific about what `frame_time` is actually used for? (Maybe reference `TimeUpdateStrategy`?)

File: crates/bevy_dev_tools/src/ci_testing.rs [hunk 2]
```diff
@@ -2,15 +2,14 @@
 
 use bevy_app::{App, AppExit, Update};
 use bevy_ecs::{
-    entity::Entity,
-    prelude::{resource_exists, Resource},
-    query::With,
-    schedule::IntoSystemConfigs,
-    system::{Local, Query, Res, ResMut},
+    entity::Entity, event::Event, prelude::Resource, query::With, system::Local, world::World,
```
Comment (BD103) [inline crates/bevy_dev_tools/src/ci_testing.rs:5]: Can this be simplified to just import from the prelude?

```suggestion
    prelude::*,
```

File: crates/bevy_dev_tools/src/ci_testing.rs [hunk 3]
```diff
@@ -61,39 +56,57 @@ pub(crate) fn setup_app(app: &mut App) -> &mut App {
...
+
+    let events = std::mem::take(&mut config.events);
+    let (to_run, remaining): (Vec<_>, _) = events
+        .into_iter()
+        .partition(|event| event.0 == *current_frame);
+    config.events = remaining;
+
+    for event in to_run {
+        debug!("Sending event: {}", event.1);
+        match event.1.as_str() {
```
Comment (BD103) [inline crates/bevy_dev_tools/src/ci_testing.rs:82]:
```suggestion
        // All custom events are handled here.
        match event.1.as_str() {
```

**Review by BD103 — APPROVED** [review]
I have a few small requests, but nothing blocking! 👍

**Comment by IceSentry** [comment]
Does this mean we could add events like moving a cursor, a mouse press or a button press? That would be useful to test some things like for example switching to each SSAO quality levels and take a screenshot of each.

**Comment by mockersf** [comment]
Yup that's the plan!

**Review by IceSentry — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
@mockersf CI failure seems related and persistent, if not particularly clear.

**Comment by mockersf** [comment]
oops, forgot the wasm runner case
@alice-i-cecile should be fixed now

**Pull request merged**